addons/mrp/migrations/10.0.2.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,424 @@
+mrp          / mrp.production           / move_finished_ids (one2many)  : NEW relation: stock.move
+# DONE: pre-migration: Renamed fields.
+
+mrp          / mrp.production           / date_planned_finished (datetime): NEW
+mrp          / mrp.production           / move_created_ids (one2many)   : DEL relation: stock.move
+mrp          / mrp.production           / move_lines2 (one2many)        : DEL relation: stock.move
+mrp          / mrp.production           / move_prod_id (many2one)       : DEL relation: stock.move
+# NOTHING TO DO
+
+mrp          / mrp.production           / picking_type_id (many2one)    : NEW relation: stock.picking.type, required: required, req_default: function
+# DONE: Pre-migration: pre-created the field
+# DONE: Post-migration: Updated all mrp.production records trying to find
+# first the most suitable picking_type_id based on the destination location,
+# and fall back to the default picking type search if none found previously.
+
+mrp          / mrp.production           / procurement_group_id (many2one): NEW relation: procurement.group
+# NOTHING TO DO
+
+mrp          / mrp.production           / state (selection)             : selection_keys is now '['cancel', 'confirmed', 'done', 'planned', 'progress']' ('['cancel', 'confirmed', 'done', 'draft', 'in_production', 'ready']')
+# DONE: Pre-migration: Copy-column. Post-migration. Mapped some values
+
+mrp          / mrp.production           / procurement_ids (one2many)    : NEW relation: procurement.order
+mrp          / mrp.production           / product_lines (one2many)      : DEL relation: mrp.production.product.line
+mrp          / mrp.production           / propagate (boolean)           : NEW
+mrp          / mrp.production           / scrap_ids (one2many)          : NEW relation: stock.scrap
+mrp          / mrp.production           / website_message_ids (one2many): DEL relation: mail.message
+mrp          / mrp.production           / workcenter_lines (one2many)   : relation is now 'mrp.workorder' ('mrp.production.workcenter.line')
+# NOTHING TO DO
+
+mrp          / mrp.production           / product_uom (many2one)        : was renamed to product_uom_id [nothing to do]
+mrp          / mrp.production           / workcenter_lines (one2many)   : was renamed to workorder_ids [nothing to do]
+# DONE: pre-migration: Renamed fields.
+
+
+mrp          / mrp.routing              / workcenter_lines (one2many)   : was renamed to operation_ids [nothing to do]
+# DONE: pre-migration: Renamed field.
+
+
+mrp          / mrp.routing.workcenter   / batch (selection)             : NEW required: required, selection_keys: ['no', 'yes'], req_default: function
+mrp          / mrp.routing.workcenter   / batch_size (float)            : NEW
+mrp          / mrp.routing.workcenter   / cycle_nbr (float)             : DEL required: required, req_default: function
+mrp          / mrp.routing.workcenter   / hour_nbr (float)              : DEL required: required, req_default: function
```

Comment:
Shouldn't this be migrated to time_cycle_manual?

Reply:
how? what is the relation?

Reply:
I did `UPDATE mrp_routing_workcenter SET time_mode='manual', time_cycle_manual=hour_nbr * 60;`, because old data was entered in terms of manual prediction, without any "10 last work orders average" kind of logic, but that might be OK to leave time_mode to automatic (I just knew my customer wants manual left in place). So I think it should be hour_nbr * 60, but it might need to be multiplied or divided by cycle_nbr or something to get the same conceptual result. For me cycle_nbr data was always 1, so I didn't look into it deeply, but the migration from hour_nbr was critical.